Src/BootCamp.Chapter/Inventory.cs

```diff
@@ -15,12 +18,45 @@ public Inventory()
 
         public Item[] GetItems(string name)
         {
-            return new Item[0];
+            var foundItems = new Item[1];
+            foreach (var item in _items)
+            {
+                if (item.GetName() == name)
+                {
+                    foundItems[0] = item;
+                }
+            }
+
+            return foundItems;
         }
 
         public void AddItem(Item item)
         {
+            _items = AppendToArray(item);
+        }
 
+        private Item[] AppendToArray(Item item)
+        {
+            var newItemArray = new Item[_items.Length + 1];
+            for (int i = 0; i < _items.Length - 1; i++)
+            {
+                newItemArray[i] = _items[i];
+            }
+            newItemArray[^1] = item;
+            return newItemArray;
+        }
+
+        internal Item SearchItem(string name)
+        {
+            foreach (var item in _items)
+            {
+                if (item.GetName() == name)
+                {
+                    return item; 
+                }
+            }
+
+            return new Item("", 0.00M, 00F); 
```

Comment:
This looks evil and error prone to me.

Reply:
why. 
It iterates through the collection and if it's find it  , it returns the object otherwise I return a sort of empty item because I need to return a Item

Reply:
Is there then a better way to remove a item if iI only know the name.
I only can find the remove method in the array class but that one needs a item and not a name,
So can you give me  some hints to get out of this.  Did the whole morning a lot of googling and trying but not nearby a better solution then I use now.

Reply:
I could do something like : 
``` 
for (var i = 0;  i > collection.Length -1; i ++) 
{
   if (collection[i].GetName() == name 
   {
   }
```  

but how do I know if I can copy the item at the same index or do I have to copy it a index before it.

Reply:
Nope, cannot be working  Inventory is not a string[] so it does not have a Length property.